pixeltable/utils/pydantic.py

```diff
@@ -0,0 +1,62 @@
+import datetime
+import typing
+from enum import Enum
+from typing import Any, Union
+
+import pydantic
+
+
+def is_json_convertible(model: type[pydantic.BaseModel]) -> bool:
+    """
+    Determine if instances of a Pydantic model can be converted to valid JSON
+    based on the type hints of its fields.
+    """
+    type_hints = typing.get_type_hints(model)
+    return all(_type_is_json_convertible(field_type) for field_type in type_hints.values())
+
+
+def _type_is_json_convertible(type_hint: Any) -> bool:
+    """Recursively check if a type hint represents a JSON-compatible type."""
+    if type_hint is type(None):
+        return True
+    if type_hint is Any:
+        return False
+
+    if type_hint in (str, int, float, bool):
+        return True
+
+    # datetime types serialize to ISO strings
+    if type_hint is datetime.datetime or type_hint is datetime.date:
+        return True
+    # UUID serializes to string
+    if hasattr(type_hint, '__name__') and type_hint.__name__ == 'UUID':
+        return True
+    # Enums serialize to their values
+    if isinstance(type_hint, type) and issubclass(type_hint, Enum):
+        return True
+
+    if isinstance(type_hint, type) and issubclass(type_hint, pydantic.BaseModel):
+        return is_json_convertible(type_hint)
+
+    origin = typing.get_origin(type_hint)
+    args = typing.get_args(type_hint)
+
+    if origin is Union:
```

Comment:
I'm not certain this will work for a Python 3.10 union (`T | U`), let's make sure there's a test case for that (I believe `T | U` maps to `types.UnionType`, not `typing.Union`)